File: examenfinal-featuredos/Core/Entities/Producto.cs
namespace Core.Entities;

public class Producto
{
    public int Id { get; set; }
    public string Nombre { get; set; } = null!;
    public string? Descripcion { get; set; }
    public decimal Precio { get; set; }
    public int Stock { get; set; }
    public bool EstaActivo { get; set; } = true;
    public int CategoriaId { get; set; }
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public string? UrlImagen { get; set; }
    public Categoria? Categoria { get; set; }

    public ICollection<Orden> Ordenes { get; set; } = new List<Orden>();
}

File: frondfinal-main/src/app/interfaces/entities/producto.ts
import { Categoria } from './categoria';
import { Orden } from './orden';

export interface Producto {
  id: number;
  nombre: string;
  descripcion?: string;
  precio: number;
  stock: number;
  estaActivo: boolean;
  categoriaId: number;
  fechaCreacion: Date;
  urlImagen?: string;
  categoria?: Categoria;
  ordenes?: Orden[];
}